packages/js-moi-providers/dist/base-provider.js

```diff
@@ -884,14 +888,28 @@ class BaseProvider extends abstract_provider_1.AbstractProvider {
         return new Promise(async (resolve, reject) => {
             let intervalId;
             let timeoutId;
+            const clearTimers = () => {
+                clearInterval(intervalId);
+                clearTimeout(timeoutId);
+            };
             const checkReceipt = async () => {
                 try {
                     const receipt = await this.getInteractionReceipt(interactionHash);
-                    if (receipt) {
+                    if (receipt == null) {
+                        return;
+                    }
+                    clearTimers();
+                    const result = this.processReceipt(receipt);
+                    const error = js_moi_manifest_1.ManifestCoder.decodeException(result.error);
+                    if (error == null) {
                         resolve(receipt);
-                        clearInterval(intervalId);
-                        clearTimeout(timeoutId);
+                        return;
                     }
+                    const err = new js_moi_utils_1.CustomError(error.data, js_moi_utils_1.ErrorCode.ACTION_REJECTED, {
+                        ...error,
+                        receipt,
+                    });
+                    reject(err);
                 }
                 catch (err) {
```

Comment:
either remove the try catch block if not required. Otherwise add a reject statement within the catch block.

Reply:
8b747eb9b883d5f22023d50be0221686f6aba2d8